lib/Controller/PersonController.php

```diff
@@ -69,46 +70,49 @@ public function index() {
 		$resp['enabled'] = $userEnabled;
 		$resp['clusters'] = array();
 
-		if ($userEnabled === 'true') {
-			$persons = $this->personMapper->findAll($this->userId);
-			foreach ($persons as $person) {
-				$personFaces = $this->faceMapper->findFacesFromPerson($this->userId, $person->getId(), $model);
-				if ($notGrouped === 'false' && count($personFaces) <= 1)
-					continue;
+		if ($userEnabled !== 'true')
+			return new DataResponse($resp);
 
-				$limit = 14;
-				$faces = [];
-				foreach ($personFaces as $personFace) {
-					if ($limit-- === 0)
-						break;
-
-					$image = $this->imageMapper->find($this->userId, $personFace->getImage());
-					$fileUrl = $this->getRedirectToFileUrl($image->getFile());
-					if (NULL === $fileUrl)
-						continue;
-
-					$face = [];
-					$face['thumb-url'] = $this->getThumbUrl($personFace->getId());
-					$face['file-url'] = $fileUrl;
-					$faces[] = $face;
-				}
+		$persons = $this->personMapper->findAll($this->userId);
+		foreach ($persons as $person) {
+			$personFaces = $this->faceMapper->findFacesFromPerson($this->userId, $person->getId(), $model);
+			if ($notGrouped === 'false' && count($personFaces) <= 1)
+				continue;
+
+			$limit = 14;
+			$faces = [];
+			foreach ($personFaces as $personFace) {
+				if ($limit-- === 0)
+					break;
 
-				$cluster = [];
-				$cluster['name'] = $person->getName();
-				$cluster['count'] = count($personFaces);
-				$cluster['id'] = $person->getId();
-				$cluster['faces'] = $faces;
+				$image = $this->imageMapper->find($this->userId, $personFace->getImage());
+				$fileUrl = $this->getRedirectToFileUrl($image->getFile());
+				if (NULL === $fileUrl) {
+					$limit++;
```

Comment:
consider moving `if ($limit-- === 0)` at the end, instead of doing ++/-- with $limit:)